src/irmin-pack/irmin_pack.ml

```diff
@@ -194,7 +194,11 @@ module Atomic_write (K : Irmin.Type.S) (V : Irmin.Hash.S) = struct
     (aux [@tailcall]) 0L @@ fun () ->
     { cache; index; block; w = watches; lock = Lwt_mutex.create () }
 
-  let unsafe_v = with_cache ~clear:unsafe_clear ~v:unsafe_v "store.branches"
+  let unsafe_v =
+    let (`Staged v) =
```

Comment:
this took me a while to convince me it's correct ; I think it will be simpler to read if you write `let (`Staged v`) ...` as a toplevel sentence.

Reply:
This means changing its type to require a unit to be passed, but I guess it's OK since `unsafe_v` isn't being exposed.

Reply:
Now changed to be `Staged` at the top-level, with the small caveat of changed type, [here](https://github.com/mirage/irmin/pull/805/files#diff-1beb7e2142825f197755316a39d00699R197).